File: synapse-transport/src/main/java/com/synapse/transport/feature/model/dto/queue/ModelTrainingJob.java
package com.synapse.transport.feature.model.dto.queue;

import java.util.UUID;

import com.synapse.transport.common.enums.ExperimentTaskType;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ModelTrainingJob {
    private UUID workspaceId;
    private UUID experimentId;
    private ExperimentTaskType taskType;
    private String datasetConfigStorageKey;
    private String datasetStorageKey;
    private String modelIrStorageKey;
    private String modelConfigStorageKey;
}

File: synapse-runtime/common/models/queue/model_training_job.py
from uuid import UUID
from pydantic import BaseModel, Field
from common.enums.task_type import TaskType


class ModelTrainingJob(BaseModel):
    workspace_id: UUID = Field(alias="workspaceId")
    experiment_id: UUID = Field(alias="experimentId")
    task_type: TaskType = Field(alias="taskType")
    dataset_config_storage_key: str = Field(alias="datasetConfigStorageKey")
    dataset_storage_key: str = Field(alias="datasetStorageKey")
    model_ir_storage_key: str = Field(alias="modelIrStorageKey")
    model_config_storage_key: str = Field(alias="modelConfigStorageKey")
